cmd/bosun/sched/check_test.go

```diff
@@ -258,6 +258,68 @@ Loop:
 	}
 }
 
+// TestCheckNotifyUnknownDefault tests the default unknownTemplate.
+func TestCheckNotifyUnknownDefault(t *testing.T) {
+	s := new(Schedule)
+	nc := make(chan string, 1)
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		b, _ := ioutil.ReadAll(r.Body)
+		nc <- string(b)
+	}))
+	defer ts.Close()
+	u, err := url.Parse(ts.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	c, err := conf.New("", fmt.Sprintf(`
+		template t {
+			subject = template
+		}
+		notification n {
+			post = http://%s/
+		}
+		alert a {
+			template = t
+			critNotification = n
+			crit = 1
+		}
+	`, u.Host))
+	if err != nil {
+		t.Fatal(err)
+	}
+	c.StateFile = ""
+	err = s.Init(c)
+	if err != nil {
+		t.Fatal(err)
+	}
+	r := &RunHistory{
+		Events: map[expr.AlertKey]*Event{
+			expr.NewAlertKey("a", opentsdb.TagSet{"h": "x"}): {Status: StUnknown},
+			expr.NewAlertKey("a", opentsdb.TagSet{"h": "y"}): {Status: StUnknown},
+		},
+	}
+	s.RunHistory(r)
+	s.CheckNotifications()
+	gotExpected := false
+Loop:
+	for {
+		select {
+		case r := <-nc:
```

Comment:
This looks deadlocky to me. Does the server complete the request before it sends to the channel? Otherwise it seems it is waiting for the client to move on to the receive before the send can complete. The whole thing seems a bit fragile, and the timeout feels odd.